macosx/PowerManager.mm

```diff
@@ -0,0 +1,176 @@
+// This file Copyright © Transmission authors and contributors.
+// It may be used under the MIT (SPDX: MIT) license.
+// License text can be found in the licenses/ folder.
+
+#import "PowerManager.h"
+
+#include <os/log.h>
+
+@interface PowerManager ()
+
+@property(nonatomic) BOOL fIsListening;
+
+@property(nonatomic) id<NSObject> fNoNapActivity;
+@property(nonatomic) id<NSObject> fNoSleepActivity;
+
+- (void)systemWillSleep:(NSNotification*)notification;
+- (void)systemDidWakeUp:(NSNotification*)notification;
+
+- (void)powerStateDidChange:(NSNotification*)notification NS_AVAILABLE_MAC(12_0);
+
+@end
+
+@implementation PowerManager
+{
+    os_log_t log;
+}
+
++ (instancetype)shared
+{
+    static PowerManager* sharedInstance = nil;
+    static dispatch_once_t onceToken;
+    dispatch_once(&onceToken, ^{
+        sharedInstance = [[PowerManager alloc] init];
+    });
+    return sharedInstance;
+}
+
+- (instancetype)init
+{
+    if ((self = [super init]))
+    {
+        log = os_log_create("org.transmission", "power");
+        _fIsListening = NO;
+        _fNoNapActivity = nil;
```

Comment:
Doesn't ARC autoset all references to nil? Or is this a requirement of our linter (it's been a while since I ran it, sorry)

Reply:
I just wanted to be explicit in initialization.
Removed anyway in 8831988fd3ac8ac44a7d45cffe68ea58c61e065d